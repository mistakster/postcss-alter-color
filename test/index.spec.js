const process = require('./utils/process');

const source = `
body {
  background: white;
  color: darkslategray;
}

#article {
  border: solid 1px rgba(47, 79, 79, 0.4);
}

.p {
  border: solid 1px #2f4f4f;
  background: rgb(47, 79, 79) 0 0 no-repeat;
}

circle {
  fill: #2f4f4f;
}

@media screen and (min-width: 480px) {
  body {
    color: hsla(180, 25%, 25%, 0.5);
    outline: solid hsl(180, 25%, 25%) 1px;
  }
}
`;

describe('Alter Color plugin', () => {
  it('should process a complex file correctly', () => {
    const options = {
      from: 'darkslategray',
      to: '#556832',
      preserveAlphaChannel: false
    };

    return process(source, options)
      .andMatchSnapshot();
  });

  it('should process a complex file correctly while preserving alpha channel', () => {
    const options = {
      from: 'darkslategray',
      to: '#556832',
      preserveAlphaChannel: true
    };

    return process(source, options)
      .andMatchSnapshot();
  });
});
