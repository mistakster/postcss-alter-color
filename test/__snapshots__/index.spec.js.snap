// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alter Color plugin should process a complex file correctly 1`] = `
"
body {
  background: white;
  color: #556832;
}

#article {
  border: solid 1px rgba(47,79,79,0.4);
}

.p {
  border: solid 1px #556832;
  background: #556832 0 0 no-repeat;
}

circle {
  fill: #556832;
}

@media screen and (min-width: 480px) {
  body {
    color: hsla(180,25%,25%,0.5);
    outline: solid #556832 1px;
  }
}
"
`;

exports[`Alter Color plugin should process a complex file correctly while preserving alpha channel 1`] = `
"
body {
  background: white;
  color: #556832;
}

#article {
  border: solid 1px rgba(85,104,50,0.4);
}

.p {
  border: solid 1px #556832;
  background: #556832 0 0 no-repeat;
}

circle {
  fill: #556832;
}

@media screen and (min-width: 480px) {
  body {
    color: rgba(85,104,50,0.5);
    outline: solid #556832 1px;
  }
}
"
`;
